core/modules/xrdsvc/SsiProvider.cc

```diff
@@ -0,0 +1,148 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/// Implement XrdSsiProviderServer to provide Qserv's SsiService
+/// implementation. Link this file when building a plugin to be used as
+/// ssi.svclib or oss.statlib.
+
+// Class header
+#include "xrdsvc/SsiProvider.h"
+
+// System headers
+#include <memory>
+#include <sstream>
+#include <sys/types.h>
+
+// Third party headers
+#include "XrdSsi/XrdSsiCluster.hh"
+#include "XrdSsi/XrdSsiLogger.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "global/ResourceUnit.h"
+#include "wpublish/ChunkInventory.h"
+#include "xrdsvc/XrdName.h"
+
+/******************************************************************************/
+/*                               G l o b a l s                                */
+/******************************************************************************/
+
+XrdSsiProvider *XrdSsiProviderServer = 
+                new  lsst::qserv::xrdsvc::SsiProviderServer;
+
+XrdSsiProvider *XrdSsiProviderLookup = XrdSsiProviderServer;
+  
+/******************************************************************************/
+/*                                  I n i t                                   */
+/******************************************************************************/
+  
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
+          char const*   parms, int argc, char **argv) {
+
+    lsst::qserv::xrdsvc::XrdName x;
+
+// Establish our instance name (many different qservs may be running here).
+//
```

Comment:
I think the style throughout qserv code is to keep comments indented with code, and it's recommended (as SHOULD) by coding standard.